### PR TITLE
Add JEI integration

### PR DESCRIPTION
**Pull request by kevinthegreat1**
Works. Custom recipe category and everything.

### CODE REVIEW

**Comment by InkEcau** [comment]
Hi! Are you working on adding more categories? I'm considering coding for this. If I do, should I just submit a pull request to the master branch?

**Comment by kevinthegreat1** [comment]
Yes, I am working on adding jei integration. By more categories do you mean other recipes other than crafting? You are welcome to contribute.

**Comment by InkEcau** [comment]
Yes! This is my current work in progress.(There are more descriptions about the recipe to be added.)
java_bkLmfw6mtU
java_jUeuICS4G7

**Comment by viciscat** [comment]
JEI 1.20.4 is out in beta btw

**Comment by aiden-powers** [comment]
would someone be so gang as to test this again if JEI is available

**Comment by aiden-powers** [comment]
Actually now that I think about it, does this make JEI a hard dependency? This might delay Skyblocker updates until JEI is available if that's the case.

**Comment by kevinthegreat1** [comment]
> Actually now that I think about it, does this make JEI a hard dependency? This might delay Skyblocker updates until JEI is available for each new version if that's the case.

No this does not.

File: src/main/java/de/hysky/skyblocker/utils/datafixer/ItemStackComponentizationFixer.java [hunk 1]
```diff
@@ -48,15 +49,16 @@ public static String componentsAsString(ItemStack stack) {
...
-			Optional<NbtElement> encodedComponent = dataComponentType.getCodec().encodeStart(nbtRegistryOps, entry.getValue().orElseThrow()).result();
+			ComponentType<?> componentType = entry.getKey();
+			Identifier componentId = Registries.DATA_COMPONENT_TYPE.getId(componentType);
+			if (componentId == null) return null;
 
-			if (componentId == null || encodedComponent.isEmpty()) {
-				return null;
-			}
+			Optional<?> component = entry.getValue();
+			if (component.isEmpty()) return "!" + componentId;
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/utils/datafixer/ItemStackComponentizationFixer.java:57]: I don't think the game can parse this? what are these changes for

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/de/hysky/skyblocker/utils/datafixer/ItemStackComponentizationFixer.java (hunk 1 above)
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/utils/datafixer/ItemStackComponentizationFixer.java:57]: I copied this from `ItemStackArgument#asString`.

**Review by AzureAaron — APPROVED** [review]
looks good

**Pull request merged**